common/test/acceptance/pages/lms/dashboard.py

```diff
@@ -157,6 +158,30 @@ def get_course_social_sharing_widget(self, widget_name):
         """ Retrieves the specified social sharing widget by its classification """
         return self.q(css='a.action-{}'.format(widget_name))
 
+    def get_courses(self):
+        """
+        Get all courses shown in the dashboard
+        """
+        return self.q(css='ul.listing-courses .course-item')
+
+    def get_course_date(self, course_number):
+        """
+        Get course date of the given course number from dashboard
+        """
+        for course in self.get_courses():
+            try:
+                course_id = course.find_element_by_css_selector('.info-course-id').text.strip()
+                course_date = course.find_element_by_css_selector('.info-date-block').text.strip()
+
+                if course_id == course_number:
```

Comment:
I think we should find an element where `course_id` is present and match it. Matching `course_number` is not accurate sometimes.